src/components/menu/menu.tsx

```diff
@@ -150,28 +179,181 @@ export class Menu {
                             'has-grid-layout': this.gridLayout,
                         }}
                     >
-                        <limel-menu-list
-                            class={{
-                                'has-grid-layout has-interactive-items':
-                                    this.gridLayout,
-                            }}
-                            items={this.items}
-                            type="menu"
-                            badgeIcons={this.badgeIcons}
-                            onSelect={this.handleSelect}
-                            ref={this.setListElement}
-                        />
+                        {this.renderBreadcrumb()}
+                        {this.renderMenuList()}
                     </limel-menu-surface>
                 </limel-portal>
             </div>
         );
     }
 
-    public componentDidRender() {
-        const slotElement = this.host.shadowRoot.querySelector('slot');
-        slotElement.assignedElements().forEach(this.setTriggerAttributes);
+    @Watch('items')
+    protected itemsWatcher() {
+        this.currentItems = null;
+
+        this.setFocus();
+    }
+
+    @Watch('currentItems')
+    protected currentItemsWatcher() {
+        this.setFocus();
+    }
+
+    @Watch('open')
+    protected openWatcher() {
+        if (!this.open) {
+            this.currentItems = null;
+
+            return;
+        }
+
+        this.setFocus();
     }
 
+    private renderBreadcrumb = () => {
+        if (!this.currentSubMenu) {
+            return;
+        }
+
+        const breadCrumbItems: MenuCrumbItem[] = [];
+        let currentItem = this.currentSubMenu;
+        while (currentItem) {
+            breadCrumbItems.unshift({
+                text: currentItem.text,
+                icon: {
+                    name: currentItem.icon,
+                    color: currentItem.iconColor,
+                },
+                menuItem: currentItem,
+            });
+            currentItem = currentItem.parentItem;
+        }
```

Comment:
Having to redo this loop on every render feels a little painful. I'm not saying we necessarily need to optimise it _now_, but its good to be aware of whenever you are in any piece of code that is part of the render-loop, as that means the code will be run pretty incessantly, and will have a disproportionate performance impact.